org.freedesktop.Sdk.Extension.openjdk21.yaml

```
@@ -0,0 +1,205 @@
+        dest-filename: apache-maven-bin.tar.gz
+        sha512: 706f01b20dec0305a822ab614d51f32b07ee11d0218175e55450242e49d2156386483b506b3a4e8a03ac8611bae96395fd5eec15f50d3013d5deed6d1ee18224
+        x-checker-data:
+          type: anitya
+          project-id: 1894
+          stable-only: true
+          url-template: http://archive.apache.org/dist/maven/maven-3/$version/binaries/apache-maven-$version-bin.tar.gz
+    build-commands:
+      - mkdir -p $FLATPAK_DEST/maven
+      - tar xf apache-maven-bin.tar.gz --strip-components=1 --exclude=jansi-native
+        --directory=$FLATPAK_DEST/maven
+      - ln -s $FLATPAK_DEST/maven/bin/mvn $FLATPAK_DEST/maven/bin/mvnDebug $FLATPAK_DEST/bin
+  - name: gradle
+    buildsystem: simple
+    cleanup:
+      - '*.bat'
+    sources:
+      - type: file
+        url: https://services.gradle.org/distributions/gradle-8.5-bin.zip
+        dest-filename: gradle-bin.zip
+        sha512: c0c07fcee0802cbdf8156475280f961ef9c0d928252642363314e25da69803fd878ab65bdc1647efa5f4b1e9b7f71e6a4daa6946d9f9923628248e9f35e8b14b
+        x-checker-data:
+          type: json
+          url: https://api.github.com/repos/gradle/gradle/releases
+          version-query: map(select(.prerelease == false)) | first | .name
+          url-query: '["https://services.gradle.org/distributions/gradle-" + $version
+            + "-bin.zip"][0]'
+    build-commands:
+      - unzip -q gradle-bin.zip -d $FLATPAK_DEST
+      - mv $FLATPAK_DEST/gradle-* $FLATPAK_DEST/gradle
+      - ln -s $FLATPAK_DEST/gradle/bin/gradle $FLATPAK_DEST/bin
+  - name: scripts
+    buildsystem: simple
+    sources:
+      - type: script
+        commands:
+          - mkdir -p /app/jre/bin
+          - cd /usr/lib/sdk/openjdk21/jvm/openjdk-21
+          - cp -ra conf lib release /app/jre/
+          - rm /app/jre/lib/src.zip /app/jre/lib/ct.sym
+          - cp -ra bin/{java,keytool,rmiregistry} /app/jre/bin
+        dest-filename: install.sh
+      - type: script
+        commands:
+          - mkdir -p /app/jdk/
+          - cd /usr/lib/sdk/openjdk21/jvm/openjdk-21
+          - cp -ra bin conf include jmods lib release /app/jdk/
+        dest-filename: installjdk.sh
+      - type: script
+        commands:
+          - export JAVA_HOME=/usr/lib/sdk/openjdk21/jvm/openjdk-21
+          - export PATH=$PATH:/usr/lib/sdk/openjdk21/bin
+        dest-filename: enable.sh
+    build-commands:
+      - cp enable.sh install.sh installjdk.sh /usr/lib/sdk/openjdk21/
+  - name: appdata
+    buildsystem: simple
+    sources:
+      - type: file
+        path: org.freedesktop.Sdk.Extension.openjdk21.appdata.xml
+    build-commands:
+      - mkdir -p ${FLATPAK_DEST}/share/metainfo
+      - cp org.freedesktop.Sdk.Extension.openjdk21.appdata.xml ${FLATPAK_DEST}/share/metainfo
+      - appstream-compose --basename=org.freedesktop.Sdk.Extension.openjdk --prefix=${FLATPAK_DEST}
+        --origin=flatpak org.freedesktop.Sdk.Extension.openjdk
```

```suggestion
      - appstream-compose --basename=${FLATPAK_ID} --prefix=${FLATPAK_DEST} --origin=flatpak ${FLATPAK_ID}
```
21 was missing, this should avoid errors